test/run/check-application

```diff
@@ -0,0 +1,161 @@
+        variables. For example:
+        run_webdriver_tests(machine,
+            ["HUB=" + selenium_addr, "BASE_URL=" + base_rul, "BROWSER=firefox"]
+        )
+        Return success of the tests
+        (True: all passed, False: at least one failed)
+    """
+    cmd_parts = ["cd /root/e2e_tests;"] + env + ["npm run test"]
+    print(" ".join(cmd_parts))
+
+    machine.execute(" ".join(cmd_parts),
+                    timeout=3600,
+                    stdout=open("/dev/stdout"))
+
+
+def run_e2e(verbose, image, browser, cpus, memory, sit):
+    # settings on composer VM
+    network = testvm.VirtNetwork(0)
+    composer = testvm.VirtMachine(verbose=verbose, image=image,
+                                  networking=network.host(),
+                                  memory_mb=memory, cpus=cpus)
+    selenium = windows = None
+
+    # settings on selenium VM
+    if 'edge' in browser:
+        windows = testvm.VirtMachine(image='windows-10', verbose=verbose,
+                                     networking=network.host(),
+                                     memory_mb=4096, cpus=4)
+        # wdio.conf.js needs key word MicrosoftEdge for edge browser
+        browser = 'MicrosoftEdge'
+    else:
+        selenium = testvm.VirtMachine(image="selenium", verbose=verbose,
+                                      networking=network.host())
+    try:
+        composer.start()
+        if selenium:
+            # actually wait here, because starting selenium takes a while
+            selenium.pull(selenium.image_file)
+            selenium.start()
+            selenium.wait_boot()
+            selenium.set_address("10.111.112.10/20")
+            selenium.upload("test/run/selenium_start.sh", "/root/")
+
+            # start selenium on the server
+            selenium.execute(command="/root/selenium_start.sh")
+        elif windows:
+            selenium = windows
+            selenium.pull(selenium.image_file)
+            selenium.start()
+
+        composer.wait_boot()
+        composer.set_address("10.111.113.1/20")
+
+        if selenium and 'edge' in browser:
+            composer.dhcp_server(range=['10.111.112.10', '10.111.112.10'])
+
+        wait_for_selenium_running(composer, "10.111.112.10")
+        wait_for_composer_running(composer)
+
+        # Now actually run the tests
+        selenium_address = "10.111.112.10" if selenium else ""
+
+        base_rul = "https://10.111.113.1:9090/welder"
+        env = [
+                "HUB=" + selenium_address,
+                "BASE_URL=" + base_rul,
+                "BROWSER=" + browser
+              ]
+        success = run_webdriver_tests(composer, env=env)
+
+        # Download html report to environment $TEST_ATTACHMENTS
+        report_dir = os.environ.get("TEST_ATTACHMENTS")
+        if not report_dir:
+            report_dir = os.getcwd()
+        if not os.path.exists(report_dir):
+            os.makedirs(report_dir)
```

Comment:
You can use makedir's `exist_ok=True` flag to avoid the if (matter of taste, of course).